Source/WebKit/Shared/Cocoa/CoreIPCCVPixelBufferRef.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if PLATFORM(COCOA)
+
+#include "ArgumentCodersCocoa.h"
+#include <mach/mach.h>
+#include <wtf/MachSendRight.h>
+#include <wtf/RetainPtr.h>
+
+namespace WebKit {
+
+class CoreIPCCVPixelBufferRef {
+public:
+    CoreIPCCVPixelBufferRef(const RetainPtr<CVPixelBufferRef>& pixelBuffer)
+        : m_sendRight(sendRightFromPixelBuffer(pixelBuffer))
+    {
+    }
+
+    CoreIPCCVPixelBufferRef(RetainPtr<CVPixelBufferRef>&& pixelBuffer)
+        : m_sendRight(sendRightFromPixelBuffer(pixelBuffer))
+    {
+    }
+
+    CoreIPCCVPixelBufferRef(MachSendRight&& sendRight)
+        : m_sendRight(WTFMove(sendRight))
+    {
+    }
+
+    RetainPtr<CVPixelBufferRef> toCF() const;
+
+    MachSendRight sendRight()
```

Comment:
Could you add && after sendRight() to indicate this should only be called on a CoreIPCCVPixelBufferRef r-value?